esp/public/media/scripts/program/modules/scheduling_checks.jsx

```diff
@@ -114,3 +160,98 @@ var RefreshButton = React.createClass({
     </button>;
   },
 });
+
+/**
+ * Calls its onClick prop to reset table greying/sorting
+ */
+var ResetButton = React.createClass({
+  propTypes: {
+    onClick: React.PropTypes.func.isRequired,
+  },
+
+  render: function () {
+    return <button onClick={this.props.onClick} className="reset-button">
+      Reset
+    </button>;
+  },
+});
+
+// Modified from react-json-table example code.
+var SelectTable = React.createClass({
+  
+  propTypes: {
+    rows: React.PropTypes.array.isRequired,
+    saveState: React.PropTypes.shape({
+      greyed: React.PropTypes.object.isRequired,
+      sort: React.PropTypes.any.isRequired,
+      reverse: React.PropTypes.bool.isRequired
+    }).isRequired,
+    header: React.PropTypes.bool.isRequired,
+    columns: React.PropTypes.array,
+    updateTableState: React.PropTypes.func.isRequired
+  },
+    
+  getInitialState: function(){
+    // We will store the sorted column and whether each row is greyed out
+    return {sort: this.props.saveState.sort, greyed: this.props.saveState.greyed, reverse: this.props.saveState.reverse};
+  },  
+  render: function(){
+    this.props.updateTableState(this.state);
```

Comment:
Child passes the parent a copy of its state so that it can be resurrected properly between closing/opening the box or refreshing. I tried just saving a copy of the table once (instead of creating a new SelectTable) but it didn't seem to work; it seems like the state dies if I close the box even if the table is saved. I read something about the state being meaningless if the component isn't mounted, which if it means what I think it means it makes sense that the state died, as opposed to me just having incorrectly tried to preserve the SelectTable.